### PR TITLE
Implementation of optional guard check for hasRoles and hasAllRoles

### LINKED ISSUE #1204: No way to check against specific 'guard' for HasRole
$user->hasRole() has no way to check against a specific guard like $user->hasRole('Admin', 'api').

$user->hasPermissionTo('edit_article', 'api') does. 

roles table has a guard_name column but no way in the library code to check against it.

Thank you.

### PR DESCRIPTION
**Pull request by dennis-git**
Implementation of optional guard check for hasRoles and hasAllRoles methods inside HasRoles trait with unit tests.

Fixes #1204

### CODE REVIEW

File: src/Traits/HasRoles.php [hunk 1]
```diff
@@ -51,9 +51,9 @@ public function roles(): MorphToMany
     /**
      * Scope the model query to certain roles only.
      *
-     * @param \Illuminate\Database\Eloquent\Builder $query
-     * @param string|array|\Spatie\Permission\Contracts\Role|\Illuminate\Support\Collection $roles
-     * @param string $guard
+     * @param  \Illuminate\Database\Eloquent\Builder  $query
+     * @param  string|array|\Spatie\Permission\Contracts\Role|\Illuminate\Support\Collection  $roles
+     * @param  string  $guard
```
Comment (drbyte) [inline src/Traits/HasRoles.php]: Why all the unnecessary double spaces in all the docblocks?

File: src/Traits/HasRoles.php [hunk 2]
```diff
@@ -172,22 +172,24 @@ public function syncRoles(...$roles)
...
+    public function hasRole($roles, string $guard = null): bool
     {
         if (is_string($roles) && false !== strpos($roles, '|')) {
             $roles = $this->convertPipeToArray($roles);
         }
 
         if (is_string($roles)) {
-            return $this->roles->contains('name', $roles);
+            return $guard ? $this->roles->where('guard_name', $guard)->contains('name',
+                $roles) : $this->roles->contains('name', $roles);
```
Comment (drbyte) [inline src/Traits/HasRoles.php]: Can you rewrite these ternaries as multi-line for better readability?
```suggestion
            return $guard 
                ? $this->roles->where('guard_name', $guard)->contains('name', $roles) 
                : $this->roles->contains('name', $roles);
```

File: src/Traits/HasRoles.php [hunk 3]
```diff
@@ -196,21 +198,21 @@ public function hasRole($roles): bool
...
+                if ($this->hasRole($role, $guard)) {
                     return true;
                 }
             }
 
             return false;
         }
 
-        return $roles->intersect($this->roles)->isNotEmpty();
+        return $roles->intersect($guard ? $this->roles->where('guard_name', $guard) : $this->roles)->isNotEmpty();
```
Comment (drbyte) [inline src/Traits/HasRoles.php]: This is fine as single-line 👍

**Comment by drbyte** [comment]
Thanks. Just a few formatting things to improve.

File: src/Traits/HasRoles.php (hunk 1 above)
Comment (dennis-git) [reply src/Traits/HasRoles.php]: These double spaces are part of PSR2 code style requirements. I reverted them back to how they were originally in the new commit.

File: src/Traits/HasRoles.php (hunk 2 above)
Comment (dennis-git) [reply src/Traits/HasRoles.php]: done

File: src/Traits/HasRoles.php (hunk 3 above)
Comment (dennis-git) [reply src/Traits/HasRoles.php]: cool

**Comment by drbyte** [comment]
Thanks!

**Comment by dennis-git** [comment]
> Thanks!
Thank you!

**Pull request merged**